specification/servicelinker/ServiceLinker.Management/DryrunResource.tsp

```diff
@@ -0,0 +1,142 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+
+namespace Microsoft.ServiceLinker;
+/**
+ * a dryrun job resource
+ */
+@subscriptionResource
+@parentResource(SubscriptionLocationResource)
+model DryrunResource is Azure.ResourceManager.ProxyResource<DryrunProperties> {
+  ...ResourceNameParameter<
+    Resource = DryrunResource,
+    KeyName = "dryrunName",
+    SegmentName = "dryruns",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface DryrunResourceOps
+  extends Azure.ResourceManager.Legacy.LegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+        ...Azure.ResourceManager.Legacy.Provider,
+        ...LocationParameter,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = DryrunResource,
+        KeyName = "dryrunName",
+        SegmentName = "dryruns",
+        NamePattern = ""
+      >>
+    > {}
+
+@armResourceOperations
+interface DryrunResources {
+  /**
+   * get a dryrun job
+   */
+  getDryrun is DryrunResourceOps.Read<DryrunResource>;
+
+  /**
+   * create a dryrun job to do necessary check before actual creation
+   */
+  createDryrun is DryrunResourceOps.CreateOrUpdateAsync<DryrunResource>;
+
+  /**
+   * update a dryrun job to do necessary check before actual creation
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @patch(#{ implicitOptionality: false })
+  updateDryrun is DryrunResourceOps.CustomPatchAsync<
+    DryrunResource,
+    PatchModel = DryrunPatch,
+    Response = ArmResourceUpdatedResponse<DryrunResource> | ArmAcceptedLroResponse<LroHeaders = ArmAsyncOperationHeader<FinalResult = DryrunResource>>
+  >;
+
+  /**
+   * delete a dryrun job
+   */
+  deleteDryrun is DryrunResourceOps.DeleteSync<DryrunResource>;
+
+  /**
+   * list dryrun jobs
+   */
+  listDryrun is DryrunResourceOps.List<
+    DryrunResource,
+    Response = ArmResponse<DryrunList>
+  >;
+}
+
+@armResourceOperations
+interface LinkersOps
+  extends Azure.ResourceManager.Legacy.LegacyOperations<
+      {
+        ...ApiVersionParameter,
+
+        #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
```

Comment:
Nit: since this looks pretty standard we should add docs for it

Reply:
There's an instance below where we did add docs for the same param, why didnt we do the same here?

Reply:
docs added and suppression has been removed

Reply:
These operations have been updated to ExtensionResource.